bagua/torch_api/algorithms/decentralized.py

```diff
@@ -107,12 +108,16 @@ def hook():
 
     def init_post_optimizer_step_hook(self, bagua_module: BaguaModule):
         def hook(optimizer: torch.optim.Optimizer):
-            for bucket in bagua_module.bagua_buckets:
+            self.optimizer_step_count += 1
 
-                for tensor in bucket.tensors:
-                    tensor.bagua_mark_communication_ready()
+            if self.optimizer_step_count == len(bagua_module.bagua_optimizers):
+                for bucket in bagua_module.bagua_buckets:
```

Comment:
too hack

the intention is to mark all tensors in the current optimizer ready or mark every tensor (even if the tensor is not in any optimizer) ready?